app/routes.py

```diff
@@ -1 +1,275 @@
+
+ #respone notification to Slack
+
+    message = f"Someone just completed the task{task.title}"
+    slack_data = {'text': message, 'channel': SLACK_CHANNEL}
+    headers = {'Authorization': f'Bearer {SLACK_BOT_TOKEN}'}
+    
+    try:
+        response = requests.post('https://slack.com/api/chat.postMessage', json=slack_data, headers=headers)
+        response.raise_for_status()
+
+    except requests.exceptions.RequestException as e:
+        return jsonify({'message': 'Failed to send Slack notification: {e}'})
+
+    return jsonify({"task": task.to_dict()}), 200
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+@goal_bp.route("", methods=['POST'])
+
+# creating a goal resource
+def create_goal():
+    request_body = validate_request(Goal, request)
+
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+    
+    db.session.add(new_goal)
+    db.session.commit()
+    
+    return ({"goal":new_goal.goal_dict()}), 201
+
+@goal_bp.route("", methods=['GET'])
+def read_all_goals():
+    goals = Goal.query.all()
+        
+    goals_response = []
+    
+    for goal in goals:
+        goals_response.append({ "title": goal.title, "id": goal.goal_id})
+    
+    return jsonify(goals_response), 200
+
+@goal_bp.route("/<goal_id>", methods=['GET'])
+def read_one_goal(goal_id):
+    
+    goal = validate_model(Goal, goal_id)
+     
+    return ({"goal": goal.goal_dict()}), 200
+
+@goal_bp.route("/<goal_id>", methods=['PUT'])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title  = request_body["title"]
+    db.session.commit()
+
+    return jsonify({"goal": goal.goal_dict()}), 200
+
+@goal_bp.route("/<goal_id>", methods=['DELETE'])
+def delete_task(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return jsonify({"details":f'Goal {goal_id} "{goal.title}" successfully deleted'}), 200
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+
+def add_tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    if 'task_ids' not in request.json:
+        abort(400)
+
+    task_ids = request.json['task_ids']
+
+    tasks = Task.query.filter(Task.task_id.in_(task_ids)).all()
+
+    if len(tasks) != len(task_ids):
+        abort(400)
+
+    goal.task_ids = task_ids
+
+    for task in tasks:
+        task.goal_id = goal_id
+
+    db.session.commit()
+
+    return jsonify({'id': goal.goal_id, 'task_ids': goal.task_ids})
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_specific_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    tasks_response = []
+
+    for task in goal.tasks:
+        task_dict = task.to_dict()
+        task_dict["goal_id"] = goal.goal_id
+        tasks_response.append(task_dict)
+
+    goal_dict = goal.goal_dict()
+    goal_dict["tasks"] = tasks_response
+
+    return jsonify(goal_dict), 200
+
+
+
+
+
```

Comment:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭